src/staging/staging_main.py

```diff
@@ -87,36 +91,84 @@ def run_staging(
     check_file_exists(snapshot_path)
 
     # load and parse the snapshot data json file
-    snapdata = load_json(snapshot_path)
-    contributors_df, responses_df = spp_parser.parse_snap_data(snapdata)
-
-    # the anonymised snapshot data we use in the DevTest environment
-    # does not include data in several columns. This fix should be removed
-    # when new anonymised data is given.
-    if network_or_hdfs == "hdfs" and config["global"]["dev_test"]:
-        responses_df["instance"] = 0
-        # In the anonymised data the selectiontype is always 'L'
-        col_size = contributors_df.shape[0]
-        random.seed(seed=42)
-        contributors_df["selectiontype"] = random.choice(["P", "C", "L"], size=col_size)
-        cellno_list = config["devtest"]["seltype_list"]
-        contributors_df["cellnumber"] = random.choice(cellno_list, size=col_size)
-    StagingMainLogger.info("Finished Data Ingest...")
-
-    val.validate_data_with_schema(contributors_df, "./config/contributors_schema.toml")
-    val.validate_data_with_schema(responses_df, "./config/long_response.toml")
-
-    # Data Transmutation
-    StagingMainLogger.info("Starting Data Transmutation...")
-    full_responses = processing.full_responses(contributors_df, responses_df)
-
-    # Validate and force data types for the full responses df
-    # TODO Find a fix for the datatype casting before uncommenting
-    val.combine_schemas_validate_full_df(
-        full_responses,
-        "./config/contributors_schema.toml",
-        "./config/wide_responses.toml",
-    )
+    # Check if feather file exists in snapshot path
+    feather_path = paths["feather_path"]
+    load_from_feather = config["global"]["load_from_feather"]
+    feather_file = os.path.join(feather_path, f"{snapshot_name}.feather")
+    feather_files_exist = check_file_exists(feather_file)
+
+    # Only read from feather if feather files exist and we are on network
+    if feather_files_exist & load_from_feather:
+        # Temporary warning due to issues with reading from DevTest
+        if network_or_hdfs != "network":
+            StagingMainLogger.warning(
+                """Reading from feather not supported (due to test data) on DevTest, and
+                is untested in prod.
+                """
```

Comment:
@AnneONS given what I mentioned in the chat about feather potentially working in prod (but not DevTest), rather than blocking being able to run on hdfs, I have added a warning here explaining the risks.

Reply:
Hi again, @dpurches , I think we might have to resort to disabling for not network for now for 2 reasons:
if we have read_from_feather = True, it doesn't work (in devtest and prob the same in DAP) and secondly, we can't run it on DevTest with read_from_freather = False either, as it can't write to Feather (though this won't be problem in Prod)

Reply:
Ah very true, I remember us finding this when we had a chat last week but I forgot to add it in. Will update the code now.